File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> myHashMap = new HashMap<>();
        for(String word: strs){
            char[] arr = word.toCharArray();
            Arrays.sort(arr);
            String sortedWord = new String(arr);
            if(!(myHashMap.containsKey(sortedWord))){
                myHashMap.put(sortedWord,new ArrayList<String>());
            }
            myHashMap.get(sortedWord).add(word);
        }
        return new ArrayList<>(myHashMap.values());
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """

        my_dict = defaultdict(list) #this hashmap maps the count directly to the array of anagrams

        for i in strs:
            count = [0] * 26 # for 26 alphabets

            for j in i: # the letters in the string
                count[ord(j) - ord('a')] += 1

            my_dict[tuple(count)].append(i)
        
        return list(my_dict.values())
                 


        